### PR TITLE
possible fix for #1473

### PR DESCRIPTION
**Pull request by gcampbell-msft**
- Defaults to using the users settings in arduino.json if it's set. 
- Removes the addition of the board name to create specific temp files. Instead, if the temp file already exists, clear it out so that no builds or files are corrupted.

### CODE REVIEW

**Comment by gcampbell-msft** [comment]
#1473

File: src/arduino/arduino.ts
Comment (benmcmorran) [inline src/arduino/arduino.ts:679]: How was the output directory created before this change, and will the new way of creating the output directory cause any conflicts with the old code path?

Also, if the `INVALID_OUTPUT_PATH` constant is no longer used then it should be removed.

File: src/debug/configurationProvider.ts
Comment (benmcmorran) [inline src/debug/configurationProvider.ts:124]: I'm guessing this won't affect too many users, but just to clarify: this means that a user that frequently switches between boards will also be forced to frequently rebuild, correct?

**Review by benmcmorran — APPROVED** [review]
These changes look reasonable to me, but I'm not enough of an expert in this extension to guarantee that there aren't other unexpected side effects. Assuming you've done some manual testing, I'd also send the original issue reporter a link to the CI build to verify that the fix works for them before we merge.

Also, looks like your CI build is currently blocked on linting issues.

File: src/debug/configurationProvider.ts
Comment (gcampbell-msft) [reply src/debug/configurationProvider.ts:124]: Yes, that would be the case.

File: src/arduino/arduino.ts
Comment (gcampbell-msft) [reply src/arduino/arduino.ts:679]: Before this change, the output directory wasn't created. Specifically, it seemed that the only case this affected was when a user had an output set in their arduino.json upon opening VSCode, this would happen. I see no real reason why not to go ahead and create the folder at this point.

**Pull request merged**
